### DISCUSSION TITLE
MPD crossfade - relevant information needed

### DISCUSSION THREAD

**Discussion by chourmovs** [MusicPlayerDaemon/MPD#1839] [General]
Hi,

I'm using MPD 0.21.18 (Not possible to easily update it) and I'm going nuts trying to make crossfade active and reliable
To help me move forward I need answer to very simple question that google/doc/chatgpt resist to answer 

- To make crossfade active do I need to modify mpd.conf or "mpc crossfade xx" is sufficient ? I found contradictory pieces of information regardig this topic. 

- My loglevel is set to verbose, can I expect finding something related to crossfade in mpd.log ? I can't see nothing, how do I ensure crossfader is active or not ?  To be honest sometime I feel my songs are crossfaded but changing crossfade time does't have impact so maybe gapless playback fool me...

Thank you for your answer

Vincent

**Comment by MaxKellermann** [comment]
Cross-fading is a runtime option, i.e. your mpc command is enough. Though cross-fading is only possible between two songs with the same audio format; if that is a problem, you can force MPD to use a specific audio format with the `audio_output_format` setting.

**Reply by chourmovs** [reply]
Thank you for your quick reply, very usefull

Still, is there a log, stdout, stderr ...  etc where I could control crossfade is effective ?

**Reply by MaxKellermann** [reply]
I'm confused by your question, because you already correctly described how cross-fading is controlled.

**Reply by chourmovs** [reply]
Sorry for my poor english that induce confusion, my sentence should have been 
"Still, is there a log, stdout, stderr ... etc where I could **check** crossfade is effective ?", meaning for example a log with
```
...
17:42:36  crossfade between track 1 and track 2 started
...
17:42:46 crossfade between track 1 and track 2 finished
...
```
or something like that, because I suspect I get false positive and that crossfading doesn't really work on my system

**Reply by MaxKellermann** [reply]
MPD doesn't log anything about cross-fading, so you can only hear it or see it on the progress bar on your MPD client (after a song change, MPD will be at 10s if cross-fade is configured to 10s, for example).

**Reply by chourmovs** [reply]
Thank you you solved my problem
